docs/testing-flakes.md

```diff
@@ -0,0 +1,60 @@
+hardware. For that reason, a reasonable chunk of our test harness isn't deterministic.
+A side effect of that is that new Pull Requests advancing through the testing pipeline
+may fail with reasons that are unrelated to the changes that comprises that pull request
+but rather by some underlying components that the cluster uses to run these tests.
+Those test failures are called flakes.
+
+The following is intended as a guide for identifying those testing flakes and to
+document them as well as possible so that it's possible to come back at a later
+time and help eliminate them or lower their occurrence rate.
+
+## What to do when a PR testing fails
+
+1. Make sure the error isn't related to the change. When confirmed, it's time to investigate
+   the flake further.
+2. After initial inspection of the flake (See Tricks below), either :
+	* CREATE a new github issue to report a flake never seen before. Try to add
+          some basic info :
+		* Title starting with : "Flake:"
+		* Links to PR where encountered and to CI-operator PR diagnostic page
+                  for the failing test
+		* A brief but complete description of how it was encountered with the last
+                  or most relevant error message
+	* **OR** UPDATE with additional findings a previously created issue for that flake:
+		* Variations in errors messages from where it was encountered before
+		* Complete or partial tracing of to the source of the problem
+		* Links to other PR or external issues referencing this flake
+3. Ensure the flake issue is labeled appropriately:
+```
+/kind test-flake
+```
+4. Back to your PR page, try to get passed the testing failure:
+```
+/retest               # to schedule a new complete test run
+/test <failing test>  # to schedule only the specific test that failed
+```
+
+good luck!
+
+## Tricks to work with flakes
+
+* Take a look at the docs/testing.md file for general testing documentation.
+* When a CI test fails look at the CI-operator issued report in the PR discussion and click
+  on the "link" link on the test that failed and you will be taken to the CI-operator PR
+  diagnostic page. There's 2 links that are interesting to click on to get started with debugging.
+	* build log: This displays the complete build and execution log for the tests. This log
+          will include the error messages of what might have went wrong during testing.
+	* artifacts: This section includes all data files and logs that were generated by different
+          portions of the code and agglomerated in a predefined storage folder ($ARTIFACT_DIR). You
+          will find information about the state of :
+		* pods
+		* events
+		* controller manager
+* In complex situations, it is possible to add custom data to the artifact directory for later
+  harvesting. Look at ./hack/artifacts.sh.
+* Note that it's generally better to use Geneva logging instead of the artifact directory for new
```

Comment:
this is condensing multiple separate points:
1) for lightweight debugging, add log entries.  For heavyweight debugging where artifacts are required, use artifacts.  Be judicious either way.
2) currently, some logs are stored as artifacts but are also available in geneva.  Also some artifacts are a crutch because actually log lines are sufficient to diagnose.  We will reduce artifacts over time, and it is recommended to exercise the geneva debugging muscles